utils_cv/tracking/model.py

```diff
@@ -243,9 +279,48 @@ def save(self, path) -> None:
         save_model(path, self.epoch, self.model, self.optimizer)
         print(f"Model saved to {path}")
 
-    def evaluate(self, results, gt) -> pd.DataFrame:
-        pass
+    def evaluate(self,
+                 results: Dict[int, List[TrackingBbox]],
+                 gt_root_path: str) -> pd.DataFrame:
+        
+        """ eval code that calls on 'motmetrics' package in referenced FairMOT script, to produce MOT metrics on inference, given ground-truth.
+        Args:
+            results: prediction results from predict() function, i.e. Dict[int, List[TrackingBbox]] 
+            gt_root_path: path of dataset containing GT annotations in MOTchallenge format (xywh)
+        Returns:
+            strsummary: pandas.DataFrame output by method in 'motmetrics', containing metrics scores
+        Raises:
+            Exception: if both `results` and `self.stored_predictions` are None.
+        """
+        if results is None:
+            if not self.stored_predictions: #TODO: add stored_predictions in predict() 
+                raise Exception("No predict() function run on dataset for for evaluation")
+            results = self.stored_predictions     
+        
+        result_filename = osp.join(gt_root_path,'results', 'results.txt')
+        if not osp.exists(osp.join(gt_root_path,'results')):
+            os.makedirs(osp.join(gt_root_path,'results'))
+        save_results_txt(results, result_filename)
+
+        #Implementation inspired from code found here: https://github.com/ifzhang/FairMOT/blob/master/src/track.py
+        evaluator = Evaluator(gt_root_path, "single_vid", "mot")
+        accs=[evaluator.eval_file(result_filename)]    
```

Comment:
Could you separate this evaluate from here? take accs as an input for another function maybe called eval_summary(). The following code should take a list of result_filenames.

Reply:
@catherine667, can you clarify more what you mean and why?

Reply:
I don't follow either.